code/game/machinery/doors/door.dm

```diff
@@ -146,6 +146,10 @@
 
 /obj/machinery/door/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
+	
+	if(istype(mover) && (mover.pass_flags & PASSMACHINE))

```

Comment:
```suggestion
	if(mover.pass_flags & PASSMACHINE)
```

Reply:
Wait, isn't this already done on the parent?

Reply:
For some reason due to the pass check already on the door, I had to add it here as well just to make sure it was checking for this flag. Not sure what a better way would be but it does work where it didn't before.